### PR TITLE
fix(theme): ignored className attribute on lazy loaded TabItem

### LINKED ISSUE #10218: TabItem ignores className if lazy attribute is set
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

When using `<TabItem className="foo">` inside a `<Tabs lazy>`, the `className` attribute is ignored. I'd expect that the behavior is the same as for `<Tabs>` (without the `lazy` attribute). 

### Reproducible demo

https://codesandbox.io/p/devbox/little-cookies-g2xysx

### Steps to reproduce

1. `lazy` to `Tabs` Item
2. `className` to `TabItem`

### Expected behavior

```tsx
<Tabs lazy>
    <TabItem value="apple" label="Apple" className={styles.apple} default>
          This is an apple 🍎
    </TabItem>
    <TabItem value="orange" label="Orange">
          This is an orange 🍊
    </TabItem>
    <TabItem value="banana" label="Banana">
          This is a banana 🍌
    </TabItem>
</Tabs>
```
with `styles.module.css`

```css
.apple {
  background-color: red;
}
```

image

### Actual behavior

The className is overwritten with `'margin-top--md'` https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-theme-classic/src/theme/Tabs/index.tsx#L125

image

### Your environment

- Public source code:
- Public site URL: https://codesandbox.io/p/devbox/little-cookies-g2xysx
- Docusaurus version used: 3.4
- Environment name and version (e.g. Chrome 89, Node.js 16.4):
- Operating system and version (e.g. Ubuntu 20.04.2 LTS):

### Self-service

- [X] I'd be willing to fix this bug myself.

### PR DESCRIPTION
**Pull request by lebalz**
Fixes #10218 

## Pre-flight checklist

- [ ] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

### Test links

Deploy preview: https://deploy-preview-_____--docusaurus-2.netlify.app/

### CODE REVIEW

**Comment by slorber** [comment]
Thanks 👍

File: packages/docusaurus-theme-classic/src/theme/Tabs/index.tsx [hunk 1]
```diff
@@ -122,7 +123,9 @@ function TabContent({
       // fail-safe or fail-fast? not sure what's best here
       return null;
     }
-    return cloneElement(selectedTabItem, {className: 'margin-top--md'});
+    return cloneElement(selectedTabItem, {
+      className: clsx('margin-top--md', className),
+    });
```
Comment (slorber) [inline packages/docusaurus-theme-classic/src/theme/Tabs/index.tsx:128]: Can you prove me that it works thanks to a visible example in the preview?

I suspected the code to not work and it doesn't look like it works

To me you should do `selectedTabItem.props.className` instead

**Comment by lebalz** [comment]
> Doesn't seem to work
> 
> See repro using swizzle:
> 
> https://stackblitz.com/edit/github-ez2gpv?file=src%2Fpages%2Findex.tsx,src%2Fpages%2Findex.module.css,src%2Ftheme%2FTabs.tsx,src%2Ftheme%2Fstyles.module.css

Don't know how i messed up my setup that i had the impression it worked - sorry for the trouble.

With your hint, it now works, and i extended the dogfooding `/pages/tabs-tests/` that shows the new behaviour:

https://deploy-preview-10219--docusaurus-2.netlify.app/tests/pages/tabs-tests/

**Comment by slorber** [comment]
Thanks 👍

**Pull request merged**